activation/handler.go

```diff
@@ -497,17 +504,47 @@ func (h *Handler) HandleGossipAtx(ctx context.Context, peer p2p.Peer, msg []byte
 }
 
 func (h *Handler) handleAtx(ctx context.Context, expHash types.Hash32, peer p2p.Peer, msg []byte) error {
-	receivedTime := time.Now()
 	var atx types.ActivationTx
 	if err := codec.Decode(msg, &atx); err != nil {
 		return fmt.Errorf("%w: %w", errMalformedData, err)
 	}
-
-	atx.SetReceived(receivedTime.Local())
+	atx.SetReceived(time.Now().Local())
 	if err := atx.Initialize(); err != nil {
 		return fmt.Errorf("failed to derive ID from atx: %w", err)
 	}
 
+	// Check if processing is already in progress
+	h.inProgressMu.Lock()
+	if sub, ok := h.inProgress[atx.ID()]; ok {
+		ch := make(chan error, 1)
+		h.inProgress[atx.ID()] = append(sub, ch)
+		h.inProgressMu.Unlock()
+		h.log.WithContext(ctx).With().Debug("atx is already being processed. waiting for result", atx.ID())
+		select {
+		case err := <-ch:
+			h.log.WithContext(ctx).With().Debug("atx processed in other task", atx.ID(), log.Err(err))
+			return err
```

Comment:
I'm not sure if returning the same `error` for all gossiped copies of the same ATX is correct? Could it be that an ATX fails validation now but might pass later (e.g. missing dependencies are now available)? 
Also if `error` is not `nil` that would be suspicious - that would mean multiple nodes gossiped the ATX and consider it valid.

Would it make sense not to return `err` here but instead continue validation as normal? If the ATX was processed already it will exit early with `errKnown` otherwise it will try to validate again.

Reply:
> Could it be that an ATX fails validation now but might pass later (e.g. missing dependencies are now available)?

I'm not sure about that. The peer that gossiped the invalid ATX should have all its dependencies already. Some transient networking issues could cause it.

I don't think it makes sense to disregard the error and retry validation. There is little chance it will succeed the second time. That ATX should be re-gossiped (if it comes from "the owner") or we will fetch it when needed.

Reply:
> The peer that gossiped the invalid ATX should have all its dependencies already.

Yes but the peer that receives the ATX and validates it might not and then fail the validation, when it could pass it at a later point (after receiving the dependencies for it).

Reply:
The node that processes a gossiped ATX proactively fetches its dependencies here: https://github.com/spacemeshos/go-spacemesh/blob/20194e8e76441a12376b5dda36abed9d381bff42/activation/handler.go#L563

It means that the peer that gossiped the ATX doesn't have the deps if that fails, which means something about that node is utterly wrong.

Reply:
Actually, there is no guarantee that `FetchReferences()` would ask the same peer that gossiped the ATX. It will ask a random peer, so a fetch request could fail if a given peer doesn't have the dep yet. This is suboptimal and could be improved in the future. 

Anyway, the _fetcher_ has a retry mechanism built-in - it will retry up to 100 times by default. It seems unlikely for `FetchReferences()` to fail if the dep exists somewhere to be fetched. Wdyt?